### PR TITLE
rewrite urlUtils.isSupportedProtocol(close #463)

### LINKED ISSUE #463: Script error on http://sidex.ru/
https://testcafe-hhhm.devexpress.com/history?timestamp=1457550274507&siteUrl=http://sidex.ru/
We are trying to proxy unsupported protocol - `viber://`.

### PR DESCRIPTION
**Pull request by miherlosev**
/cc @churkin @LavrovArtem 

Also I return some tests that we lost during move code to the github.

### CODE REVIEW

**Comment by inikulin** [comment]
And onccceee again, can we change isSupportedProtocol so it will a)lookup for the protocol presence (it's not hard) b)Check if protocol is `http://`, `https://` or `//`? Speaking clearly: use whitelists instead of blacklists

**Comment by miherlosev** [comment]
`isSupportedProtocol` is used for absolute and relative urls.
For relative url whitelist is a bad solution.
For example:
`file.html` - supported protocol.
`file://html` - unsupported protocol.
`..index.html` - supported protocol.
`..%2Fdata` - supported protocol.

File: src/utils/url.js [hunk 1]
```diff
@@ -209,7 +197,12 @@ export function parseUrl (url) {
 }
 
 export function isSupportedProtocol (url) {
-    return !/^\s*(chrome-extension:|blob:|javascript:|about:|mailto:|tel:|data:|skype:|skypec2c:|file:|#)/i.test(url);
+    url = url || '';
+    url = trim(url);
+
+    var protocol = url.match(PROTOCOL_RE);
+
+    return !protocol || /^(http:|https:|\/\/)/i.test(protocol[0]);
```
Comment (inikulin) [inline src/utils/url.js]: Shouldn't it be 
```js
/^((http:|https:)?\/\/)/i
```
?
BTW we should move it to the const.

File: test/client/fixtures/utils/url-test.js [hunk 2]
```diff
@@ -42,6 +41,44 @@ test('resolveUrlAsDest', function () {
...
+test('isSupportedProtocol', function () {
+    ok(!urlUtils.isSupportedProtocol('JavaScript:0;'));
+    ok(!urlUtils.isSupportedProtocol('javascript:0;'));
+    ok(!urlUtils.isSupportedProtocol('skypec2c://r/202'));
+    ok(!urlUtils.isSupportedProtocol('chrome-extension://google.com/image.png'));
+    ok(!urlUtils.isSupportedProtocol(' data:asdasdasdasdasdasd'));
+
+    // NOTE: GH-463
+    ok(!urlUtils.isSupportedProtocol('viber://public?id=test'));
+    ok(!urlUtils.isSupportedProtocol('whatsapp://send?text=test'));
```
Comment (inikulin) [inline test/client/fixtures/utils/url-test.js]: This lacks checks for the supported protocols

**Comment by miherlosev** [comment]
FPR

File: test/client/fixtures/utils/url-test.js [hunk 3]
```diff
@@ -42,6 +41,38 @@ test('resolveUrlAsDest', function () {
     strictEqual(urlUtils.resolveUrlAsDest('http://g.tbcdn.cn/??kissy/k/1.4.2/seed-min.js'), 'http://g.tbcdn.cn/??kissy/k/1.4.2/seed-min.js');
 });
 
+test('isSupportedProtocol', function () {
+    ok(!urlUtils.isSupportedProtocol('#42'));
+    ok(!urlUtils.isSupportedProtocol(' data:asdasdasdasdasdasd'));
+    ok(!urlUtils.isSupportedProtocol('chrome-extension://google.com/image.png'));
```
Comment (inikulin) [inline test/client/fixtures/utils/url-test.js]: I've asked to add tests for all scenarios and positive scenarios as well:
`http://example.org`,
`https://example.org`,
`//example.org`,
`/some/path`
`somepath`
etc.

**Comment by inikulin** [comment]
\r-

**Comment by inikulin** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Pull request merged**